src/controllers/dotController.js

```diff
@@ -0,0 +1,73 @@
+const dotRepo = require('../repository/dotRepository.js');
+
+const positionSubscribers = [];
+const dotIds = {};
+
+async function subscribe(req, res) {
+  let { scriptId } = req.params;
+
+  res.writeHead(200, {
+    'Cache-Control': 'no-cache',
+    'Content-Type': 'text/event-stream',
+    'Connection': 'keep-alive',
+  });
+
+  const id = Math.random();
+
+  req.on('close', () => {
+    console.log(`Connection closed: ${id}`);
+    positionSubscribers.filter((subscriber) => subscriber.id !== id);
+  });
+
+  // Push res to subscribers, so we can send events as they come in.
+  positionSubscribers.push({ id, scriptId, res });
+
+  // send initial data for all actor positions, updates will be sent later as new positions are received
+  const dots = await dotRepo.getAll(scriptId);
```

Comment:
Does this need a try catch?

Reply:
yes, adding

Reply:
Thanks for catching this! 
I've been encountering occasional errors that then errored out because the request had already sent headers and meant to address that. Doing so had slipped my mind.